plugin/db/util/driverutil.go

```diff
@@ -132,151 +132,178 @@ type MigrationExecutor interface {
 // Returns the created migraiton history id and the updated schema on success.
 func ExecuteMigration(ctx context.Context, l *zap.Logger, executor MigrationExecutor, m *db.MigrationInfo, statement string) (migrationHistoryID int64, updatedSchema string, resErr error) {
 	var prevSchemaBuf bytes.Buffer
-	// Don't record schema if the database hasn't exist yet.
-	if !m.CreateDatabase {
+	if m.CreateDatabase {
+		// Don't record schema if the database hasn't exist yet.
+	} else if m.Type == db.Baseline {
+		// Record the updated schema of the previous migration if exists.
+		prevMigrationHistoryFind := db.MigrationHistoryFind{
+			Database: &m.Database,
+			Limit:    new(int),
+		}
+		*prevMigrationHistoryFind.Limit = 1
+		migrationHistoryList, err := executor.FindMigrationHistoryList(ctx, &prevMigrationHistoryFind)
+		if err != nil {
+			l.Error("Failed to find migration history", zap.Error(err))
+			return -1, "", err
+		}
+		if len(migrationHistoryList) > 0 {
+			prevSchemaBuf.WriteString(migrationHistoryList[0].Schema)
```

Comment:
It's intentional that we do NOT fetch the stored prev schema, and use the actual live schema instead, because

1. Schema can drift https://bytebase.com/blog/what-is-database-schema-drift

2. Our baseline could have a bug (like before baseline unexpected executes the DDL).  By capturing the real schema before and after migration, we can see if the baseline does work as expected.

It looks a bit confusing from the UI, so I think we should optimize the UI instead.

Reply:
The original design to capture the before and after schema is for sanity checking the actual migration change.

Before schema + Migration change = After schema.

Fetching the last stored migration schema could break this equation in cases mentioned above.

Reply:
I see. I gonna try to change UI instead